### ISSUE TITLE
Completions for method overrides in @code blocks don't include the return type, braces, and access level modifier

### ISSUE THREAD

**Issue by danroth27** [aspnet/AspNetCore#13319]
Completions for method overrides in @code blocks don't include the return type, braces, and access level modifier.

Repro steps:
- Create a new Blazor app
- In a component .razor file add a `@code` block
- Type `override ` and then complete with `OnInitialized`

Actual result:

```csharp
override OnInitialized()
```

Expected result:

```csharp
protected override void OnInitialized()
{
    base.OnInitialized();
}
```

@NTaylorMullen

**Comment by NTaylorMullen** [comment]
The gist is that the insertions are occurring in the C# projection buffer and aren't being mapped correctly back to Razor. Depending on how the edits are formed this may or may not be fixable by us.

**Comment by pranavkm** [comment]
@NTaylorMullen could you investigate and figure out if a fix for this can go in 16.4?

**Comment by NTaylorMullen** [comment]
Investigated and it wasn't what I initially thought. Roslyn doesn't support complex completions in projected scenarios. Basically [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs,140) they end up looking for information about the current document (they only allow C#) and fail because the "current" document happens to be Razor. To make this work Roslyn would need to expand their custom completion support to work in Razor scenarios by re-mapping the completion trigger location (Razor) to the active `subjectBuffer` (C#). This also assumes that manual completions "just work" after doing this.

@dpoeschl is this a known limitation / any thoughts on the above?

**Comment by ivanbasov** [comment]
Thank you for investigations and explanations, @NTaylorMullen ! I have created an issue on the Roslyn side: https://github.com/dotnet/roslyn/issues/38596 Let us see what we can do.

**Comment by NTaylorMullen** [comment]
This was fixed externally for 16.4-preview2 with a slight caveat that formatting is slightly broken after completing an override statement. Closing for now since the core blocker of this is fixed.

Also to note, re-formatting the document will fix the miscellaneous formatting bug.
